### DISCUSSION TITLE
Disabling the captcha while authenticating

### DISCUSSION THREAD

**Discussion by frodoe7** [invertase/react-native-firebase#4183] [Q&A] [answered]
I cannot find a way to disable the captcha

here's the code

```
    firebase
      .auth()
      .signInWithPhoneNumber(phoneNumber, captchaVerifier)
```

how to edit this above code to hide / disable the captcha

I found this reference but not sure how to achieve it in RN App

https://firebase.google.com/docs/auth/web/phone-auth

**Comment by andersonaddo** [comment]
https://rnfirebase.io/auth/phone-auth

**Comment by mikehardy** [comment]
You'll need to test on a real device, as well. The simulator usually captchas, the real device it does not happen to me after setup

**Comment by frodoe7** [comment]
@andersonaddo @mikehardy 

even if I managed to disable the captcha
it still open a popup with a web view

I don't need this view to be popped up at all

**Comment by frodoe7** [comment, marked as answer]
I revised Firebase official docs
unfortunately , looks it is not possible to disable the captcha

https://firebase.google.com/docs/auth/web/phone-auth#set-up-the-recaptcha-verifier

**Reply by ghost** [reply]
Hi @frodoe7, see my comment regarding Android SafetyNet. Pretty easy to verify phone number without popup+reCAPTCHA at all on Android https://github.com/invertase/react-native-firebase/discussions/4183#discussioncomment-1041328

I hope it helps! Works perfectly for me.

**Reply by bhaktitud** [reply]
what about ios @o-vexler ?

**Comment by taschik** [comment]
I'm wondering how do you implement end-to-end tests that are testing e.g. a login flow for a user, if occasionally the simulator will pop a recaptcha for you to select some boats or bridges?

**Reply by mikehardy** [reply]
@taschik docs:

https://firebase.google.com/docs/auth/web/phone-auth#integration-testing

https://rnfirebase.io/reference/auth#settings
https://rnfirebase.io/reference/auth/authsettings#appVerificationDisabledForTesting

**Reply by taschik** [reply]
Thanks @mikehardy !

**Comment by ghost** [comment]
Hi @frodoe7 , successfully disabled the reCAPTCHA verification (without popup!) using **[Android SaftyNet:](https://developer.android.com/training/safetynet)**

**Just follow the 2 simple steps in the following link and it'll work (Android):** 
https://firebase.google.com/docs/auth/android/phone-auth#enable-app-verification

> If a user has a device with Google Play Services installed, and Firebase Authentication can verify the device as legitimate with Android SafetyNet, phone number sign-in can proceed.

Tested on:

- real device
- @react-native-firebase/app@12.2.0
- @react-native-firebase/auth@12.2.0

**Reply by cavishek39** [reply]
@o-vexler can you please write down the steps of how to configure this Android SafetyNet for a React-Native project, since I've done all the steps mentioned above, but how to get rid of the re-captcha UI when trying to verify a mobile number?

**Reply by SahilPrabhakar7** [reply]
You have  enabled Android Device Verification from  GCP for your project . 
https://console.cloud.google.com/apis/library/androidcheck.googleapis.com

**Reply by cavishek39** [reply]
@SahilPrabhakar7 yes, I did that. But I have a question will this method gonna work in firebase SDK or not? Since I am not using

- @react-native-firebase/app@12.2.0
- @react-native-firebase/auth@12.2.0 
this two libraries.

**Reply by SahilPrabhakar7** [reply]
@cavishek39 for this method to work you have to use these two libraries.I had the same issue and had to installed these two libraries for this method to work.

**Reply by cavishek39** [reply]
Okay, thanks @SahilPrabhakar7 for your help.

**Reply by mikehardy** [reply]
But please please please use up to date versions before logging any problems :pray:  https://invertase.io/blog/react-native-firebase-versioning - I work really hard in the release notes to leave the exact steps needed to update, and usually it's quite easy
